### ISSUE TITLE
Sort::link(), other directions

### ISSUE THREAD

**Issue by zelenin** [yiisoft/yii2#8456]
The Sort widget does not support the generation of links with two different directions (opposite or default direction only). It's understandable In the GridView, but Sort is not part of GridView. So expected functionality is creating two different links like:

``` php
public function link($attribute, $direction = null, $options = []) {
if ($direction !== null) {
// generate link with $direction
} else {
//old behavior
}
}
```

https://github.com/yiisoft/yii2/blob/3a1e0f3a5c6f75ed08bb8cea321c2391045bc804/framework/data/Sort.php#L295

**Comment by cebe** [comment]
a generated link normally uses default direction when it was not sorted by this attribute and other direction if the current page does sort by this attribute. Can you explain how you want to use this?

**Comment by zelenin** [comment]
```
Sort by price: <?= $sort->link('price', SORT_ASC, ['label' => 'Up']) ?> | <?= $sort->link('price', SORT_ASC, ['label' => 'Down']) ?>
```

**Comment by toir427** [comment]
provider
```
new ActiveDataProvider([
                                          'query'      => $query,
                                          'sort'       => [
                                              'defaultOrder' => ['name' => SORT_ASC],
                                              'attributes'   => [
                                                  'name'    => [
                                                      'label' => 'Name (A-Z)',
                                                  ],
                                                  '-name'   => [
                                                      'label' => 'Name (Z-A)',
                                                  ],
                                              ],
                                          ],
                                      ]);
```
yii\widgets\LinkSorter file extends and using this code to get dropdownlist
```
protected function renderSortLinks()
    {
        $selected   = [];
        $ddOptions  = ['onchange' => 'window.location=this.value'];
        $attributes = empty($this->attributes) ? $this->sort->attributes : $this->attributes;

        $label = '';
        $a     = \Yii::$app->request->get('sort') ?: key($this->sort->defaultOrder);
        if (strpos($a, '-') == 0) {
            $label = $attributes[$a]['label'];
            unset($attributes[$a]);
        }


        $items[''] = $label;
        foreach ($attributes as $attribute => $label) {
            $url         = $this->sort->createUrl($attribute);
            $items[$url] = $label['label'];
        }

        return Html::dropDownList('sort', $selected, $items, $ddOptions);
    }
```

![sorter](https://user-images.githubusercontent.com/19358691/35723745-3ed202b0-081d-11e8-90ad-a90087616061.png)

```
<select name="sort" onchange="window.location=this.value">
<option value="">Name (A-Z)</option>
<option value="/category/smartphones?sort=-name">Name (Z-A)</option>
</select>
```

**Comment by samdark** [comment]
Won't be implemented in 2.0.

**Comment by vaso2** [comment]
I've actually implemented it, backward compatible, and works like this:
```
<?= $sort->link('price', ['class' => 'sort', 'label' => 'Cheap first', 'direction' => SORT_ASC]) ?>
<?= $sort->link('price', ['class' => 'sort', 'label' => 'Expensive first', 'direction' => SORT_DESC]) ?>
<?= $sort->link('promo', ['class' => 'sort', 'label' => 'Promo']) ?>
```
I can submit PR, if such feature is needed by others.
